File: app/src/main/java/com/roman/kubik/spivanyklicejista/presentation/main/di/MainModule.java
package com.roman.kubik.spivanyklicejista.presentation.main.di;

import android.app.Activity;

import com.roman.kubik.spivanyklicejista.general.di.ActivityScope;
import com.roman.kubik.spivanyklicejista.presentation.main.MainActivity;
import com.roman.kubik.spivanyklicejista.presentation.main.MainContract;

import dagger.Module;
import dagger.Provides;

/**
 * {@link MainActivity} module
 * Created by kubik on 1/14/18.
 */
@Module
public class MainModule {

    private MainActivity activity;

    public MainModule(MainActivity activity) {
        this.activity = activity;
    }

    @Provides
    @ActivityScope
    public MainContract.View getView() {
        return activity;
    }

    @Provides
    @ActivityScope
    public Activity getActivity() {
        return activity;
    }

}

File: app/src/main/java/com/roman/kubik/spivanyklicejista/presentation/main/di/MainModule.kt
package com.roman.kubik.spivanyklicejista.presentation.main.di

import com.roman.kubik.spivanyklicejista.domain.song.SongInteractor
import com.roman.kubik.spivanyklicejista.general.di.ActivityScope
import com.roman.kubik.spivanyklicejista.presentation.main.MainActivity
import com.roman.kubik.spivanyklicejista.presentation.main.MainContract
import com.roman.kubik.spivanyklicejista.presentation.main.MainPresenter
import dagger.Module
import dagger.Provides
import io.reactivex.disposables.CompositeDisposable

@Module
class MainModule(private val activity: MainActivity) {

    @Provides
    @ActivityScope
    fun getView(): MainContract.View = activity

    @Provides
    @ActivityScope
    fun getPresenter(songInteractor: SongInteractor, compositeDisposable: CompositeDisposable): MainContract.Presenter = MainPresenter(activity, songInteractor, compositeDisposable)

}
